packages/grid/x-data-grid/src/hooks/features/pagination/gridPaginationInterfaces.ts

```diff
@@ -1,38 +1,30 @@
+import { GridPaginationModel } from '../../../models/gridPaginationProps';
+
 export interface GridPaginationState {
-  pageSize: number;
-  page: number;
-  pageCount: number;
-  rowCount: number;
+  paginationModel: GridPaginationModel;
 }
 
 export interface GridPaginationInitialState {
-  pageSize?: number;
-  page?: number;
+  paginationModel?: Partial<GridPaginationModel>;
 }
 
 /**
- * The apiRef methods handled by `useGridPageSize`
+ * The pagination API interface that is available in the grid `apiRef`.
  */
-export interface GridPageSizeApi {
+export interface GridPaginationApi {
+  /**
+   * Sets the displayed page to the value given by `page`.
+   * @param {number} page The new page number.
+   */
+  setPage: (page: number) => void;
```

Comment:
Mark as `@deprecated`?

Reply:
Yes that would be more logical to do and I wanted to do that too but this [concern](https://github.com/mui/mui-x/pull/7147#discussion_r1061251536) (second part of the comment) also makes sense that most frequent use-case for the users would be updating either of the `page` or `pageSize`, so keeping these helpers kind of makes the API easy to use. What do you think?